### PR TITLE
Update pywebsocket to latest master

### PR DESCRIPTION
**Pull request by jgraham**
We don't seem to be able to [get a new release](https://github.com/GoogleChromeLabs/pywebsocket3/issues/30), and pywebsocket bugs are currently blocking Python 3.12 support

### CODE REVIEW

**Review by Ms2ger — APPROVED** [review]
fun.

File: tools/wptserve/setup.py [hunk 1]
```diff
@@ -1,9 +1,13 @@
 from setuptools import setup
 
+import os
+
+here = os.path.dirname(__file__)
+
 PACKAGE_VERSION = '4.0.1'
```
Comment (whimboo) [inline tools/wptserve/setup.py:7]: @jgraham I assume that you will bump and release in a follow-up PR or can we combine the release here as well?

**Review by whimboo — COMMENTED** [review]
Looks fine to me as well. Just one question regarding the release. I assume that packaging works fine when it gets build and uploaded to PyPI?

File: tools/wptserve/third_party/pywebsocket3/pywebsocket3/memorizingfile.py
Comment (WeizhongX) [inline tools/wptserve/third_party/pywebsocket3/pywebsocket3/memorizingfile.py:1]: This file has shebang but is not executable. Can we remove that?

**Comment by WeizhongX** [comment]
I see some test files (e.g. https://github.com/web-platform-tests/wpt/blob/master/websockets/handlers/echo_wsh.py) are still referencing `mod_pywebsocket`. Is that right?

I tried to roll the change to chromium, and run the change on our CI. It reported some websocket tests are failing. 
https://chromium-review.googlesource.com/c/chromium/src/+/5438911?tab=checks

**Comment by foolip** [comment]
@jgraham https://github.com/GoogleChromeLabs/pywebsocket3/issues/30 is fixed now, if you want to use pypi instead.

**Pull request closed without merging**